Birthday/Modules/Home/ViewModel/HomeViewModel.swift

```diff
@@ -0,0 +1,45 @@
+
+import Apollo
+import Combine
+import Foundation
+
+final class HomeViewModel: ObservableObject {
+  
+  @Published var birthdays: [BirthdayModel] = []
+  @Published var isLoading: Bool = false
+  @Published var error: GraphQLError? = nil
+
+  private var cancellables: Set<AnyCancellable> = []
+  
+  let birthdayRepository: BirthdaysRepository
+  
+  init(birthdayRepository: BirthdaysRepository) {
+    self.birthdayRepository = birthdayRepository
+    getBirthdays()
+  }
+  
+  private func getBirthdays() {
+    self.isLoading = true
+    birthdayRepository.getBirthdays()
+      .sink { [weak self] result in
+        DispatchQueue.main.async { [weak self] in
+          self?.isLoading = false
+        }
+        switch result {
+        case .finished: break
+        case .failure(let error):
+          DispatchQueue.main.async {
+            self?.error = error as? GraphQLError
+            print(error)
```

Comment:
you can only print error and remove "@Published var error: GraphQLError? = nil" it used only here